version: "3.8"

services:

  db:
    image: postgres:15
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - "POSTGRES_HOST_AUTH_METHOD=trust"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 30s
      timeout: 30s
      retries: 5
  
  app:
    build: .
#    command: bash -c "./manage.py makemigrations && ./manage.py migrate && ./manage.py migrate --run-syncdb && ./manage.py runserver 0.0.0.0:8000"
    command: bash -c "./manage.py makemigrations && ./manage.py migrate && ./manage.py migrate --run-syncdb && python manage.py collectstatic --no-input && gunicorn PyEditorial.wsgi:application --bind 0.0.0.0:8000"
    volumes:
      - app_static:/code
    ports:
      - "8000:8000"
    depends_on:
      - db
volumes:
  postgres_data:
  app_static: